zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -28,19 +32,25 @@ pub(crate) struct BeginTxGadget<F> {
     tx_nonce: Cell<F>,
     tx_gas: Cell<F>,
     tx_gas_price: Word<F>,
-    mul_gas_fee_by_gas: MulWordByU64Gadget<F>,
+    gas_fee: MulWordByU64Gadget<F>,
     tx_caller_address: Cell<F>,
     tx_caller_address_is_zero: IsZeroGadget<F>,
     tx_callee_address: Cell<F>,
     tx_is_create: Cell<F>,
     tx_value: Word<F>,
+    effective_tx_value: Word<F>,
+    effective_gas_fee: Word<F>,
     tx_call_data_length: Cell<F>,
     tx_call_data_gas_cost: Cell<F>,
+    nonce: Cell<F>,
+    nonce_prev: Cell<F>,
     reversion_info: ReversionInfo<F>,
-    sufficient_gas_left: RangeCheckGadget<F, N_BYTES_GAS>,
+    gas_not_enough: LtGadget<F, N_BYTES_GAS>,
```

Comment:
The spec used `MAX_N_BYTES = 31`, but `N_BYTES_GAS` seems to be a more reasonable value. So after the circuit PR is reviewed and merged in we should remember to update the spec.

Reply:
Sounds good